homie/src/lib.rs

```diff
@@ -512,6 +576,55 @@ mod tests {
         drop(rx);
     }
 
+    #[tokio::test]
+    async fn sleep_then_ready_again_succeeds() {
+        let (mut device, rx) = make_test_device();
+
+        device.start().await.unwrap();
+        device.ready().await.unwrap();
+        device.sleep().await.unwrap();
+        device.ready().await.unwrap();
+
+        // Need to keep rx alive until here so that the channel isn't closed.
+        drop(rx);
+    }
+
+    #[tokio::test]
+    async fn alert_then_ready_again_succeeds() {
+        let (mut device, rx) = make_test_device();
+
+        device.start().await.unwrap();
+        device.ready().await.unwrap();
+        device.alert().await.unwrap();
+        device.ready().await.unwrap();
+
+        // Need to keep rx alive until here so that the channel isn't closed.
+        drop(rx);
+    }
+
+    #[tokio::test]
+    async fn disconnect_succeeds_before_ready() {
+        let (mut device, rx) = make_test_device();
+
+        device.start().await.unwrap();
+        device.disconnect().await.unwrap();
+
+        // Need to keep rx alive until here so that the channel isn't closed.
+        drop(rx);
+    }
+
+    #[tokio::test]
+    async fn disconnect_succeeds_after_ready() {
+        let (mut device, rx) = make_test_device();
+
+        device.start().await.unwrap();
+        device.ready().await.unwrap();
+        device.disconnect().await.unwrap();
```

Comment:
(optional) Tests can return result in the same way that main() can return result. This would make your tests feel more like real code.

Reply:
Done.